config/fcos/v1_6_exp/translate.go

```diff
@@ -367,3 +374,79 @@ func buildGrubConfig(gb Grub) string {
 	superUserCmd := fmt.Sprintf("set superusers=\"%s\"\n", strings.Join(allUsers, " "))
 	return "# Generated by Butane\n\n" + superUserCmd + strings.Join(cmds, "\n") + "\n"
 }
+
+func (c Config) handleSelinux(options common.TranslateOptions) (types.Config, translate.TranslationSet, report.Report) {
+	rendered := types.Config{}
+	ts := translate.NewTranslationSet("yaml", "json")
+	var r report.Report
+
+	for i, module := range c.Selinux.Module {
+		yamlPath := path.New("yaml", "selinux", "module", i)
+		if module.Name != "" && module.Content != "" {
+			rendered = processModule(rendered, module, options, ts, r, yamlPath)
+		} else {
+			r.AddOnWarn(yamlPath, common.ErrFieldInvalid)
+		}
+	}
+
+	if len(rendered.Storage.Files) != 0 {
+		newFilesystem := types.Filesystem{
+			Device: "/dev/disk/by-label/boot",
+			Format: util.StrToPtr("ext4"),
+			Path:   util.StrToPtr("/boot"),
+		}
+
+		rendered.Storage.Filesystems = append(rendered.Storage.Filesystems, newFilesystem)
+		ts.AddFromCommonSource(path.New("yaml", "selinux", "module", 0), path.New("json", "storage"), newFilesystem)
+
+	}
+
+	return rendered, ts, r
+}
+
+func processModule(rendered types.Config, module Module, options common.TranslateOptions, ts translate.TranslationSet, r report.Report, yamlPath path.ContextPath) types.Config {
+	src, compression, err := baseutil.MakeDataURL([]byte(module.Content), nil, !options.NoResourceAutoCompression)
+	if err != nil {
+		r.AddOnError(yamlPath, err)
+		return rendered
+	}
+
+	// Create module file
+	modulePath := fmt.Sprintf("/etc/selinux/targeted/modules/active/extra/%s.cil", module.Name)
+	newFile := types.File{
+		Node: types.Node{
+			Path: modulePath,
+		},
+		FileEmbedded1: types.FileEmbedded1{
+			Append: []types.Resource{
+				{
+					Source:      util.StrToPtr(src),
+					Compression: compression,
+				},
+			},
+		},
+	}
+
+	filePath := path.New("json", "storage", "files", len(rendered.Storage.Files), newFile)
+	rendered.Storage.Files = append(rendered.Storage.Files, newFile)
+	ts.AddFromCommonSource(yamlPath, filePath, newFile)
+
+	// Create systemd unit to import module
+	cmdToExecute := "/usr/sbin/semodule -i" + modulePath
+	newUnit := types.Unit{
+		Name:    "selinux-import-module@.service",
+		Enabled: util.BoolToPtr(true),
+		Dropins: []types.Dropin{
+			{
+				Name:     module.Name + ".conf",
+				Contents: util.StrToPtr(" [Unit]\n" + "Description=Import SELinux module\n" + "[Service]\n" + "Type=oneshot\n" + "RemainAfterExit=no\n" + "ExecStart=" + cmdToExecute + "\n" + "[Install]\n" + "WantedBy=multi-user.target\n"),
```

Comment:
Should include the module name in the description.
Also, shouldn't `RemainAfterExit` be `yes`?

Reply:
I agree. Thanks!